### PR TITLE
[Fix](delete command) Mark delete sign when do delete command in MoW table

### LINKED ISSUE #34551: [Bug] 使用delete 删除的数据仍然可以查询到
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0.4

### What's Wrong?

--1. 先创建TEST1表
drop table IF EXISTS TEST1;
CREATE TABLE IF NOT EXISTS `TEST1` (
  `ID` int NULL COMMENT 'ID',
  `ID1` int NULL COMMENT 'ID1',
	`ID2` int NULL COMMENT 'ID2'
) UNIQUE KEY(
  `ID`
) DISTRIBUTED BY HASH(`ID`) BUCKETS AUTO PROPERTIES (
  "replication_allocation" = "tag.location.default: 1",
  "enable_unique_key_merge_on_write" = "true",
	"store_row_column" = "true"
);

-- 2. 表中插入一条记录
insert into TEST1(ID,ID1,ID2) values(1,1,1);
select * from TEST1;
 查询结果
image

-- 3. 删除此记录
delete from TEST1 where ID=1;

--4. 设置部分更新
set enable_unique_key_partial_update=true;
set enable_insert_strict=false;

--5. 在此插入相同key值记录
insert into TEST1(ID,ID1) values(1,2);
select * from TEST1;

image

### What You Expected?

上面第五步骤中期望结果为
image

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

close #34551 

Problem: As shown in the issue above, if a key deleted by a delete statement is written to by updating only certain columns, the data will not display correctly.

Reason: The delete statement deletes the data by writing a delete predicate, which is stored in the rowset meta and applied during data retrieval to filter the data. However, partial column updates do not consider the effect of the delete predicate when reading the original data. The imported key should be considered as a new key (since it has already been deleted), but it is actually treated as an old key. Therefore, only some columns are updated, leading to incorrect results.

Solution: Consider the delete predicate during partial column updates, but this method will result in reading more columns, as shown in #35766. Thus, in this PR, we change the delete operation in the mow table from writing a delete predicate to writing a delete sign, which effectively resolves the issue.

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run performance

**Comment by Yukang-Lian** [comment]
run p0

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -2619,6 +2619,15 @@ public class Config extends ConfigBase {
...
     public static String inverted_index_storage_format = "V2";
 
+    @ConfField(mutable = true, masterOnly = true, description = {
+            "是否开启delete语句写delete predicate。若开启，会提升delete语句的性能，"
+                    + "但delete后进行部分列更新可能会出现部分数据错误的情况。若关闭，会降低delete语句的性能来保证正确性。",
+            "Enable the 'delete predicate' for DELETE statements. If enabled, it will enhance the performance of "
+                    + "DELETE statements, but partial column updates after a DELETE may result in erroneous data. "
+                    + "If disabled, it will reduce the performance of DELETE statements to ensure accuracy."
+    })
+    public static boolean enable_delete_on_delete_predicate = false;
```
Comment (zhannngchen) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2629]:
```suggestion
    public static boolean enable_mow_delete_on_delete_predicate = false;
```

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 2]
```diff
@@ -2619,6 +2619,15 @@ public class Config extends ConfigBase {
     })
     public static String inverted_index_storage_format = "V2";
 
+    @ConfField(mutable = true, masterOnly = true, description = {
+            "是否开启delete语句写delete predicate。若开启，会提升delete语句的性能，"
```
Comment (zhannngchen) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2623]:
```suggestion
            "是否在unique key表mow实现上开启delete语句写delete predicate。若开启，会提升delete语句的性能，"
```

File: regression-test/suites/unique_with_mow_p0/partial_update/test_new_partial_update_delete.groovy [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            sql "set show_hidden_columns = false;"
+            sql "set enable_unique_key_partial_update=true;"
+            sql "set enable_insert_strict=false;"
+            sql "insert into ${tableName2} (k1,c1) values(1,2)"
+            // 1,2,NULL,NULL,NULL
+            qt_sql24 "select * from ${tableName2} order by k1;"
+
+
+
+            sql """alter table ${tableName2} set ("enable_delete_on_delete_predicate"="true") """
```
Comment (zhannngchen) [inline regression-test/suites/unique_with_mow_p0/partial_update/test_new_partial_update_delete.groovy:134]: add cases on mor table, mor table should not support this property.

**Comment by Yukang-Lian** [comment]
run buildall

**Review by zhannngchen — DISMISSED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run p0

**Comment by Yukang-Lian** [comment]
run cloud_p1

**Comment by Yukang-Lian** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Review by zhannngchen — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by weilai201** [comment]
那个版本发布后，修复此问题呢？

**Comment by zhannngchen** [comment]
@weilai201 2.1.5

**Pull request merged**